src/optimagic/visualization/backends.py

```diff
@@ -25,6 +27,26 @@ def __call__(
         legend_properties: dict[str, Any] | None,
         margin_properties: dict[str, Any] | None,
         horizontal_line: float | None,
+        subplot: Any | None = None,
+    ) -> Any: ...
+
+
+@runtime_checkable
+class GridLinePlotFunction(Protocol):
+    def __call__(
+        self,
+        lines_list: list[list[LineData]],
+        *,
+        n_rows: int,
+        n_cols: int,
+        titles: list[str] | None,
+        xlabel: str | None,
+        ylabel: str | None,
+        template: str | None,
+        height: int | None,
+        width: int | None,
+        legend_properties: dict[str, Any] | None,
+        margin_properties: dict[str, Any] | None,
     ) -> Any: ...
```

Comment:
Can you add a descriptive docstring to this Protocol and the LinePlotFunction Protocol?

The idea would be that given the Protocol and the docstring, it should be clear to someone implementing a new backend on what to do...